src/VisualStudio/Core/Def/FindReferences/Entries/DocumentSpanEntry.cs

```diff
@@ -185,18 +186,29 @@ protected override IList<System.Windows.Documents.Inline> CreateLineTextInlines(
                 return inlines;
             }
 
-            public override bool TryCreateColumnContent(string columnName, [NotNullWhen(true)] out FrameworkElement? content)
+            public override bool TryCreateColumnContent(string columnName, out Action? disposeCallback, [NotNullWhen(true)] out FrameworkElement? content)
             {
-                if (base.TryCreateColumnContent(columnName, out content))
+                if (base.TryCreateColumnContent(columnName, out disposeCallback, out content))
                 {
                     // this lazy tooltip causes whole solution to be kept in memory until find all reference result gets cleared.
                     // solution is never supposed to be kept alive for long time, meaning there is bunch of conditional weaktable or weak reference
                     // keyed by solution/project/document or corresponding states. this will cause all those to be kept alive in memory as well.
                     // probably we need to dig in to see how expensvie it is to support this
                     var controlService = _excerptResult.Document.Project.Solution.Services.GetRequiredService<IContentControlService>();
-                    controlService.AttachToolTipToControl(content, () =>
+                    var disposable = controlService.AttachToolTipToControl(content, () =>
                         CreateDisposableToolTip(_excerptResult.Document, _excerptResult.Span));
 
+                    disposeCallback += () =>
```

Comment:
wow.  that this is being `+=` is subtle.  Can you doc htat?  I was def confused a bit. :)

Reply:
dumb question.  i see what you're doing here, creating a side-table of disposables that go away when the snapshot goes away.  Is it not possible though to just reutrn FrameworkElements that do this cleanup once they themselves are disposed?

Reply:
📝 FrameworkElement is not disposable, and WPF does not notify on final release

Reply:
➡️ Documented the use of the `+=` operator